src/common/hostname.cc

```diff
@@ -14,12 +14,15 @@
 
 #include "common/hostname.h"
 
-#include <unistd.h>
-
 std::string ceph_get_hostname()
 {
   char buf[1024];
-  gethostname(buf, 1024);
+#ifdef HOST_NAME_MAX
```

Comment:
need to include `limits.h` for this macro, see http://pubs.opengroup.org/onlinepubs/009604499/basedefs/limits.h.html

Reply:
@tchaikov That is what I just deleted after applying what @liewegas requested.
By the way I will re-add as your request.

Reply:
no, you didn't add `#ifndef HOST_NAME_MAX`. b/c `HOST_NAME_MAX` is not define on all platforms.

Reply:
it's not addressed.